torchattacks/attacks/fmn.py

```diff
@@ -0,0 +1,338 @@
+import math
+from functools import partial
+from typing import Optional
+
+import torch
+import torch.nn as nn
+from torch import Tensor
+from torch.optim import SGD
+from torch.optim.lr_scheduler import CosineAnnealingLR
+
+from ..attack import Attack
+
+
+class FMN(Attack):
+    r"""
+    FMN in the paper 'Fast Minimum-norm Adversarial Attacks through Adaptive Norm Constraints'
+    [https://arxiv.org/abs/2102.12827]
+
+    Distance Measure : L0, L1, L2, Linf
+
+    Args:
+        model (nn.Module): The model to be attacked.
+        device: The device to run the attack on. If None, the default device will be used.
+        norm (float): The norm for distance measure. Defaults to float('inf').
+        steps (int): The number of steps for the attack. Defaults to 10.
+        alpha_init (float): The initial alpha for the attack. Defaults to 1.0.
+        alpha_final (Optional[float]): The final alpha for the attack. Defaults to alpha_init / 100 if not provided.
+        gamma_init (float): The initial gamma for the attack. Defaults to 0.05.
+        gamma_final (float): The final gamma for the attack. Defaults to 0.001.
+        starting_points (Optional[Tensor]): The starting points for the attack. Defaults to None.
+        binary_search_steps (int): The number of binary search steps. Defaults to 10.
+
+    Shape:
+        - images: (N, C, H, W) where N is the number of batches, C is the number of channels, H is the height,
+          and W is the width. It must have a range [0, 1].
```

Comment:
And there is no need to break the line here, you can copy it directly from the shape part in other files. 🤪

Reply:
Ok, thank you